change/@ni-nimble-components-99393253-07d1-4ec4-902b-41680bddd36e.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Empty list items are rendering poorly for me. If I set this markdown:
```
1. 
2. 
3. abc
4. def
```

Then I get this output:

<img width="117" alt="image" src="https://github.com/ni/nimble/assets/10500124/c49f4c69-1af6-4311-ad13-9a3f31342a91">

Reply:
The issue I encountered was not reproducible in the Chrome browser, but it can be replicated in Firefox. To create grouped lists, I set the margin of the `<p>` tag within `<li>` to 0px. However, this caused a problem in Firefox when the `<p>` tag inside `<li>` was empty. To resolve this issue, I added the following style.

```css
li > p: empty {
    display: contents;
}
```

@jattasNI After attempting to test this in various browsers, I also ran Storybook in WebKit as mentioned [here](https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#development-workflow), but the component fails to render completely. I also checked the [snapshots in the Chromatic build](https://www.chromatic.com/component?appId=60e89457a987cf003efc0a5b&csfId=incubating-rich-text-viewer&buildNumber=6332&k=64aed9cc2661cff185fba28b-1200-snapshot-true-safari&h=27&b=-13), only for Safari the component not rendered. Could you please share some information that will help us in identifying the issue?
![image](https://github.com/ni/nimble/assets/123377523/ee147f37-0329-4eb9-b99a-ad0293c0e00d)

Reply:
I can confirm that I was testing in Firefox and that the issue now appears to be fixed.

It's good that it's working fine in Safari 16.5.1. I see this too. Chromatic is likely using an older version (their docs just say 16 https://www.chromatic.com/docs/infrastructure-release-notes). If we can find an approach that works in the older version that would be ideal so that we get regression testing and so that the component works for users who are stuck on older versions. 

You'll have to do some digging to find out what that approach might be though. You could look for recently fixed webkit bug reports about accessors in constructors. https://bugs.webkit.org/ Or you could play around with different patterns for setting markdown (set it at different times, change from accessor to method, etc) and see if any of them work better.

Reply:
I tried setting the markdown using `methods` and was able to see the content being rendered in the component without any issues. [Here](https://60e89457a987cf003efc0a5b-hhuarjcekv.chromatic.com/?path=/docs/incubating-rich-text-viewer--docs) is the Chromatic build for the same. But I couldn't find any bugs in the WebKit related to accessors. 

The component renders in the WebKit as well as in [Chromatic Safari snapshots](https://www.chromatic.com/component?appId=60e89457a987cf003efc0a5b&csfId=incubating-rich-text-viewer&buildNumber=6404&k=64b542091ec8fc719d61076a-1200-snapshot-true-safari&h=4&b=-3) with the approach of using the [observables](https://www.fast.design/docs/fast-element/observables-and-state/#internal-observation) of FAST instead of accessors as per the discussion in the [other comment](https://github.com/ni/nimble/pull/1335#discussion_r1263938629).